docs/DeveloperGuide.md

```diff
@@ -1,38 +1,152 @@
 # Developer Guide
 
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* Java Coding Standard (Basic): [Official Documentation](https://se-education.org/guides/conventions/java/basic.html)
+* Code Quality (Guide): (https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html)
 
 ## Design & implementation
-
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### "Add" Command Feature
+The "AddCommand" is responsible for adding fields and records to the inventory. 
+Concurrently, "add" command is implemented with behaviour to update CSV file to persist changes made to inventory.
+This command supports the following options:
+* '-h': Add custom fields to the inventory.
+* '-d': Add records to the inventory corresponding to defined fields.
+* '-hu': Updates fields in the inventory.
+
+#### Architecture Overview
+The 'AddCommand' is part of the `command` package, and interacts with the following key components:
+* **Inventory**: Contains the current state of the inventory, including fields and records.
+* **Csv**: Handles reading from and writing to the CSV file for data storage persistence.
+* **Ui**: Handles user interactions component such as message to user during command execution.
+
+Command flow of 'AddCommand' can be visualized with the following architecture diagram:
+![Architecture Diagram](link-to-architecture-diagram)
+
+
+#### Component-Level Design
+"AddCommand" Class:
+* processes the input arguments and executes the logic based on flags (-h, -d, -hu) respectively.
+* fields and records are validated before updating inventory data into the CSV file.
+
+"AddCommand" Methods:
+* handleAddMultipleFields(): processes input fields, validate fields, and update the inventory accordingly.
+* handleAddRecord(): handles logic for adding records into the inventory. If successfully add records, data will be appended into the CSV file.
+* handleUpdateFields(): allow updating of existing fields and types in the inventory.
+  ![Class Diagram](link-to-class-diagram)
+
+#### Sequence Diagram
+*Illustrates how "AddCommand" interacts with "Inventory" and "Csv" classes when adding a record:
+
+![Sequence Diagram](link-to-sequence-diagram)
```

Comment:
You may consider adding an explanation for your diagrams, to help the reader understand the command better